### PR TITLE
Robust Deck Integration Tests for TenantIDs

### PR DESCRIPTION
**Pull request by mpherman2**
/assign @chaodaiG @cjwagner

### CODE REVIEW

File: prow/test/integration/test/deck_test.go [hunk 1]
```diff
@@ -43,3 +240,117 @@ func TestDeck(t *testing.T) {
...
+			if err = yaml.Unmarshal(body, &got); err != nil {
+				t.Fatalf("Failed unmarshal prowjobs %v", err)
+			}
+
+			if allExpected := expectedPJsInDeck(&expected, &got); !allExpected {
+				t.Fatalf("Not all expected PJs are present. want: %v\n got:%v", expected, got)
+			}
+
+			if unexpectedFound := unexpectedPJsInDeck(&unexpected, &got); unexpectedFound {
+				t.Fatalf("Not all expected PJs are present. want: %v\n got:%v", expected, got)
```
Comment (cjwagner) [inline prow/test/integration/test/deck_test.go:315]: This error message needs to be updated.

File: prow/test/integration/test/deck_test.go [hunk 2]
```diff
@@ -43,3 +240,117 @@ func TestDeck(t *testing.T) {
...
+	}
+	return res
+}
+
+func expectedPJsInDeck(pjs *prowjobv1.ProwJobList, deck *prowjobv1.ProwJobList) bool {
+	for _, expected := range getSpecs(pjs) {
+		found := false
+		for _, spec := range getSpecs(deck) {
+			if diff := cmp.Diff(expected, spec); diff == "" {
+				found = true
```
Comment (cjwagner) [inline prow/test/integration/test/deck_test.go:337]: nit: break here to make this (completely negligibly) faster.

File: prow/test/integration/test/deck_test.go [hunk 3]
```diff
@@ -17,12 +17,209 @@ limitations under the License.
...
+			Labels: map[string]string{
+				"created-by-prow": "true",
+			},
+		},
+		Spec: prowjobv1.ProwJobSpec{
+			Job: "Default TenantID",
+			ProwJobDefault: &prowjobv1.ProwJobDefault{
+				TenantID: config.DefaultTenantID,
+			},
+			Namespace: defaultNamespace,
```
[the remark this answers is not in the document]
Comment (mpherman2) [reply prow/test/integration/test/deck_test.go]: Done

File: prow/test/integration/test/deck_test.go [hunk 4]
```diff
@@ -17,12 +17,209 @@ limitations under the License.
...
+				TenantID: "tester",
+			},
+			Hidden:    true,
+			Namespace: defaultNamespace,
+		},
+		Status: prowjobv1.ProwJobStatus{
+			State:     prowjobv1.TriggeredState,
+			StartTime: v1.NewTime(time.Now()),
+		},
+	}
```
[the remark this answers is not in the document]
Comment (cjwagner) [reply prow/test/integration/test/deck_test.go:190]: Sure, separate PR sounds good.

**Review by cjwagner — APPROVED** [review]

**Pull request merged**